File: FourierShift_MakeFile/core/convolution.c
 #include<complex.h> 
#include <stddef.h>
 #define MIN(X, Y) (((X) < (Y)) ? (X) : (Y))
#define MAX(X, Y) (((X) < (Y)) ? (Y) : (X))
static void kconv(const float * in0,
const int n,
const float complex *w,float * out) {for(int i = 0; i < n ; ++i)
	    {out[i] = 
w[0]* in0[i + 0] +
w[1]* in0[i + 1] +
w[2]* in0[i + 2] +
w[3]* in0[i + 3] +
w[4]* in0[i + 4] +
w[5]* in0[i + 5] +
w[6]* in0[i + 6] +
w[7]* in0[i + 7] +
w[8]* in0[i + 8] +
w[9] * in0[i + 9] ;
	    	}
	}
void dconv(const float *in, const int n, const float complex *w, float *out){
	 kconv(in, n, w,out);
}

File: FourierShift_MakeFile/core/convolution.py
#!/usr/bin/env python3
import sys


if len(sys.argv) == 2:
	size = int(sys.argv[1])
	weights_param = ''
	in_param = 'const float * in0,\n'
	for i in range(size):
		weights_param += 'const float w'+ str(i)+',\n'

	# n_param = 'const int n,\n'
	out_param = 'float * out)'

	head = ' #include <stddef.h>\n static void kconv('+ in_param + weights_param + out_param 
	top_body = r'''{ for(int i = 0; i < ''' + str(size) + '''; ++i)
	    {\n'''

	body_params = ''
	for i in range(size - 1):
		body_params += 'w'+ str(i) +'* in0[i + '+ str(i) +'] +\n'
	body = '{out[i] = \n'
	end = r'''w'''+ str(size-1) + ''' * in0[i + ''' +str(size-1)+'''] ;
	    	}
		}
	}'''

	code_1 = head + top_body+ body + body_params + end
	print(code_1)

	head = 'void dconv(const float *in, const float *w, float*out){\n'
	params = ''

	for i in range(size):
		params += 'w[' + str(i) + '],'

	code_2 = head + '\t kconv(in,' + params + 'out);\n}'
	print(code_2)




elif len(sys.argv) == 3:
	size = int(sys.argv[1])
	weights_param = ''
	in_param = 'const float * in0,\n'
	for i in range(size):
		weights_param += 'const float w'+ str(i)+',\n'

	# n_param = 'const int n,\n'
	out_param = 'float * out)'

	head = ' #include <stddef.h>\n static void kconv('+ in_param + weights_param + out_param +';'
	code_1 = head
	print(code_1)
	code_2 = '\nvoid dconv(const float *in, const float *w, float*out);'
	print(code_2)